Library/Formula/pinentry-mac.rb

```diff
@@ -13,9 +12,35 @@ class PinentryMac < Formula
     sha1 "4d80f3954ddb16618074d9d978f3a83aedddb8db" => :mountain_lion
   end
 
+  # This is the script previously curled and immediately executed without verification by the project's Makefile.
+  resource "core" do
+    url "https://raw.githubusercontent.com/GPGTools/GPGTools_Core/0cfb8fd7ee0104a2c542a8d5593f1e252ce44bf9/newBuildSystem/prepare-core.sh"
+    sha256 "b987ebd807d9b10b2ee0be54bc87b59d22ec2997ee216b22d862ec5593dd4772"
+  end
+
+  # This is the git repo the script clones but served over a secure protocol and tied to a
+  # more controlled commit reference. It should be updated periodically. The script above
+  # is still used by the Makefile to determine correct error checking/handling.
+  resource "core_clone" do
+    url "https://github.com/GPGTools/GPGTools_Core.git",
+        :revision => "0cfb8fd7ee0104a2c542a8d5593f1e252ce44bf9"
+  end
+
   depends_on :xcode => :build
 
   def install
+    buildpath.install resource("core")
+    (buildpath/"GPGTools_Core").install resource("core_clone")
+    chmod 0755, "prepare-core.sh"
+
+    # Remote code execution with no security mechanism in place is *BAD*. Replace upstream no checks
+    # with SHA256 on the script, and remove the git clone in favour of cloning a specific
+    # commit in this brew formula (And thus tying it to Git (And Github's) SHA1 security).
```

Comment:
To be honest, I flat out didn't expect this to get much serious discussion due to the level of changes made, so I left the extensive notes in so people could follow my often weird thought process a little easier. I'll trim the comments back a bit.